app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -70,17 +83,14 @@ class DatabaseCleanupManagerImpl(
         }
     }
 
-    private fun cleanupSentInfectionMessages() {
+    private fun cleanupSentTemporaryExposureKeys() {
         launch {
-            val configuration = configurationRepository.observeConfiguration().blockingFirst()
-
-            infectionMessageDao.removeSentInfectionMessagesOlderThan(MessageType.Revoke.Suspicion, ZonedDateTime.now())
-
-            val thresholdYellowMessages = ZonedDateTime.now().minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeSentInfectionMessagesOlderThan(MessageType.InfectionLevel.Yellow, thresholdYellowMessages)
-
-            val thresholdRedMessages = ZonedDateTime.now().minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeSentInfectionMessagesOlderThan(MessageType.InfectionLevel.Red, thresholdRedMessages)
+            // The sent temporary exposure keys are removed in case we detect that we are not anymore in quarantine.
+            if (quarantineRepository.observeQuarantineState()
+                    .blockingFirst() is QuarantineStatus.Free
```

Comment:
Don't you want to introduce suspend method instead of blockingFirst?

Reply:
Updated